integration-tests/scaffolding/extrinsicHelpers.ts

```diff
@@ -29,76 +30,114 @@ type AddProviderPayload = { authorizedMsaId?: any; schemaIds?: any; expiration?:
  * up in the test.
  */
 
-
-/** Generic wrapper **/
-export async function signAndSend<T extends ISubmittableResult>(
-    f: () => SubmittableExtrinsic<"rxjs", T>,
-    keys: KeyringPair,
-    customErrorHandler: (err: RpcError, caught: Observable<T>) => Observable<T> = (err, caught) => caught): Promise<EventMap> {
-    return firstValueFrom(f().signAndSend(keys).pipe(
-        filter(({ status }) => status.isInBlock || status.isFinalized),
-        catchError((err: RpcError, caught) => {
-            console.log(`code: ${err.code}, name: ${err.name}, message: ${err.message}`);
-            if (customErrorHandler) {
-                return customErrorHandler(err, caught);
-            }
-            throw err;
-        }),
-        parseResult(),
-    ));
+export enum SubmissionType {
+    Submit,
+    GetEstimatedFee,
+    FundOperation,
 }
 
-/** Query Extrinsics */
-export async function getAccountInfo(api: ApiRx, address: string): Promise<FrameSystemAccountInfo> {
-    return firstValueFrom(api.query.system.account(address));
+export class Extrinsic<T extends ISubmittableResult = ISubmittableResult> {
+
+    private extrinsic: () => SubmittableExtrinsic<"rxjs", T>;
+    private keys: KeyringPair;
+
+    constructor(extrinsic: () => SubmittableExtrinsic<"rxjs", T>, keys: KeyringPair) {
+        this.extrinsic = extrinsic;
+        this.keys = keys;
+    }
+
+    public async signAndSend(): Promise<EventMap> {
+        return firstValueFrom(this.extrinsic().signAndSend(this.keys).pipe(
+            filter(({ status }) => status.isInBlock || status.isFinalized),
+            parseResult(),
+        ));
+    }
+
+    public async getEstimatedTxFee(): Promise<bigint> {
+        return firstValueFrom(this.extrinsic().paymentInfo(this.keys).pipe(
+            map((info) => info.partialFee.toBigInt())
+        ));
+    }
+
+    private async fundOperation(source?: KeyringPair): Promise<void> {
+        const amount = await this.getEstimatedTxFee();
+        await ExtrinsicHelper.transferFunds(source || devAccounts[0].keys, this.keys, amount).signAndSend();
```

Comment:
It looks like most extrinsics are using the default case ("//Alice"), but I don't think it is guaranteed that this account will be funded each time. We should have a method that encapsulates the account returned here.

Reply:
This method allows the caller to specify a funding source; if one is not provided it falls back to the default '//Alice' account. From my understanding, this (as well as the other dev accounts) is guaranteed to be funded in a development chain. The only real reason to provide an alternate funding source is if we suspect the amount of tests in the suite would drain the '//Alice' account if all operations were funded from there; in that case we might revisit this. But since the idea is that the integration test pipeline will always be run against a new local Frequency chain, funding should not be an issue at this point.

Reply:
Right, but there is no guarantee that `devAccounts[0]` is Alice. I'm proposing wrapping the account we use with a method. 

Granted, a method would not guarantee it either, but makes it would make it easier to decipher for the next developer.

Reply:
It doesn't matter which `devAccounts` item we use. We don't care if it's Alice. It could be Bob, Charlie, etc. The point is we just want to use one of the default dev accounts for funding purposes. It doesn't matter which one.

Reply:
I'd like to see the actual name of the account rather than an index into an array.   While it doesn't matter which account is used, it does matter that the developer knows which account it is.

Reply:
Sounds good. I'll include it in my next PR.